pandas/io/formats/format.py

```diff
@@ -974,6 +974,7 @@ def __init__(self, *args, **kwargs):
         # float_format is expected to be a string
         # formatter should be used to pass a function
         if self.float_format is not None and self.formatter is None:
+            self.fixed_width = False
```

Comment:
can you see if u can change ``fixed_width`` to a cached property instead of setting it (need to remove from the signature as well)

Reply:
is this possible?

Reply:
There is a good chance I'm not understanding the comment, but I'm not sure we can set this as a cached property (using the `cache_readonly` decorator?) since it is set in the base class. As I say I might be missing something...

Reply:
you can set it in each of the subclasses. I just don't think we actually need this. try doing this as a property first to see if it works.